grudge/reductions.py

```diff
@@ -160,8 +159,9 @@ def nodal_sum_loc(dcoll: DiscretizationCollection, dd, vec) -> float:
         return sum(nodal_sum_loc(dcoll, dd, vec[idx])
                    for idx in np.ndindex(vec.shape))
 
+    # FIXME: do not force result onto host
     actx = vec.array_context
-    return sum([actx.np.sum(grp_ary) for grp_ary in vec])
+    return sum([actx.to_numpy(actx.np.sum(grp_ary))[()] for grp_ary in vec])
```

Comment:
Why is this needed? I would have expected this to stay the way it was (with a `to_numpy` in `allreduce` above)

Reply:
* Fixed for `sum` here: https://github.com/inducer/grudge/compare/335d27cfc84f222795693d8d6d90bfab21be0247..8131936e38f02b590261198529c8d167362b9d39
* Added a better FIXME for the others in the same push